File: 프로그래머스/lv3/42884. 단속카메라/단속카메라.py
def solution(routes):
    answer = 0
    q=[]
    routes.sort(key=lambda x:x[1])
    for cur in routes:
        s=cur[0]
        e=cur[1]
        if q and q[-1]>=s:continue
        q.append(e)
    return len(q)

File: 프로그래머스/lv3/42884. 단속카메라/단속카메라.java
import java.util.*;
class Solution {
    public int solution(int[][] routes) {
        int answer = 0;
        Arrays.sort(routes, (a,b)-> Integer.compare(a[1], b[1]));
        int pos=-30001;
        for (int i=0;i<routes.length;i++)
        {
            if (pos>=routes[i][0])continue;
            pos=routes[i][1];
            answer+=1;
        }
        
        return answer;
    }
}